miio/airpurifier_miot.py

```diff
@@ -0,0 +1,411 @@
+    "filter_rfid_tag": {"siid": 14, "piid": 1},
+    "filter_rfid_product_id": {"siid": 14, "piid": 3},
+    # Other (siid=15)
+    "app_extra": {"siid": 15, "piid": 1},
+}
+
+
+class AirPurifierMiotException(DeviceException):
+    pass
+
+
+class OperationMode(enum.Enum):
+    Auto = 0
+    Silent = 1
+    Favorite = 2
+    Fan = 3
+
+
+class LedBrightness(enum.Enum):
+    Bright = 0
+    Dim = 1
+    Off = 2
+
+
+class AirPurifierMiotStatus:
+    """Container for status reports from the air purifier."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        self.filter_type_util = FilterTypeUtil()
+        self.data = data
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if device is on."""
+        return self.data["power"]
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return "on" if self.is_on else "off"
+
+    @property
+    def aqi(self) -> int:
+        """Air quality index."""
+        return self.data["aqi"]
+
+    @property
+    def average_aqi(self) -> int:
+        """Average of the air quality index."""
+        return self.data["average_aqi"]
+
+    @property
+    def humidity(self) -> int:
+        """Current humidity."""
+        return self.data["humidity"]
+
+    @property
+    def temperature(self) -> Optional[float]:
+        """Current temperature, if available."""
+        if self.data["temperature"] is not None:
+            return self.data["temperature"]
+
+        return None
+
+    @property
+    def fan_level(self) -> int:
+        """Current fan level."""
+        return self.data["fan_level"]
+
+    @property
+    def mode(self) -> OperationMode:
+        """Current operation mode."""
+        return OperationMode(self.data["mode"])
+
+    @property
+    def led(self) -> bool:
+        """Return True if LED is on."""
+        return self.data["led"]
+
+    @property
+    def led_brightness(self) -> Optional[LedBrightness]:
+        """Brightness of the LED."""
+        if self.data["led_brightness"] is not None:
+            try:
+                return LedBrightness(self.data["led_brightness"])
+            except ValueError:
+                return None
+
+        return None
+
+    @property
+    def buzzer(self) -> Optional[bool]:
+        """Return True if buzzer is on."""
+        if self.data["buzzer"] is not None:
+            return self.data["buzzer"]
+
+        return None
+
+    @property
+    def buzzer_volume(self) -> Optional[int]:
+        """Return buzzer volume."""
+        if self.data["buzzer_volume"] is not None:
+            return self.data["buzzer_volume"]
+
+        return None
+
+    @property
+    def child_lock(self) -> bool:
+        """Return True if child lock is on."""
+        return self.data["child_lock"]
+
+    @property
+    def favorite_level(self) -> int:
+        """Return favorite level, which is used if the mode is ``favorite``."""
+        # Favorite level used when the mode is `favorite`.
```

Comment:
Should this check if the mode is favorite and return None if not?

Reply:
Nope. It's still a setting that's kept stored whichever the mode currently is.